src/errors.ts

```diff
@@ -0,0 +1,26 @@
+export class SerializationError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.message = message;
+    this.name = this.constructor.name;
+    Error.captureStackTrace(this, this.constructor);
```

Comment:
All `Error.captureStackTrace` calls probably need a check for `if (Error.captureStackTrace) ...` because it's a non-standard V8 extension